### DISCUSSION TITLE
Is there a way to differentiate between errors returned by .Insert() etc?

### DISCUSSION THREAD

**Discussion by SamusAranX** [aarondl/sqlboiler#1349] [Q&A] [answered]
As part of test code, I'd like to, for example, attempt to insert a duplicate record to verify that unique constraints are working.
I have checked the documentation but couldn't find a way to figure out whether an error returned by `.Insert()` is a constraint validation error or something else. Is this currently possible?

**Comment by stephenafamo** [comment, marked as answer]
SQLBoiler wraps errors from the database driver, so you need to look at the the documentation of your driver and see how to tell the errors apart.

I wrote an article in 2021 about doing this for PostgreSQL using the `github.com/lib/pq` driver. <https://stephenafamo.com/blog/posts/handling-postgresql-errors>

There is also a wiki entry for `pgx`: https://github.com/jackc/pgx/wiki/Error-Handling

If you're using MySQL, SQLite or SQL Server, the patters will be the same, but you'd need to find the underlying error type and use `errors.As` to get that concrete error.
